tests/Bots/DirectoryTests.hs

```diff
@@ -215,10 +219,121 @@ testDelistedServiceRemoved tmp =
         cath <## "#privacy: bob removed SimpleX-Directory from the group"
         bob <# "SimpleX-Directory> SimpleX-Directory is removed from the group ID 1 (privacy)."
         bob <## ""
-        bob <## "Group is no longer listed in the directory."
+        bob <## "The group is no longer listed in the directory."
         superUser <# "SimpleX-Directory> The group ID 1 (privacy) is de-listed (directory service is removed)."
         groupNotFound cath "privacy"
 
+testDelistedRoleChanges :: HasCallStack => FilePath -> IO ()
+testDelistedRoleChanges tmp =
+  withDirectoryService tmp $ \superUser dsLink ->
+    withNewTestChat tmp "bob" bobProfile $ \bob -> do
+      withNewTestChat tmp "cath" cathProfile $ \cath -> do
+        bob `connectVia` dsLink
+        registerGroup superUser bob "privacy" "Privacy"
+        addCathAsOwner bob cath
+        groupFound cath "privacy"
+        -- de-listed if service role changed
+        bob ##> "/mr privacy SimpleX-Directory member"
+        bob <## "#privacy: you changed the role of SimpleX-Directory from admin to member"
+        cath <## "#privacy: bob changed the role of SimpleX-Directory from admin to member"
+        bob <# "SimpleX-Directory> SimpleX-Directory role in the group ID 1 (privacy) is changed to member."
+        bob <## ""
+        bob <## "The group is no longer listed in the directory."
+        superUser <# "SimpleX-Directory> The group ID 1 (privacy) is de-listed (SimpleX-Directory role is changed to member)."
+        groupNotFound cath "privacy"
+        -- re-listed if service role changed back without profile changes
+        cath ##> "/mr privacy SimpleX-Directory admin"
+        cath <## "#privacy: you changed the role of SimpleX-Directory from member to admin"
+        bob <## "#privacy: cath changed the role of SimpleX-Directory from member to admin"
+        bob <# "SimpleX-Directory> SimpleX-Directory role in the group ID 1 (privacy) is changed to admin."
+        bob <## ""
+        bob <## "The group is listed in the directory again."
+        superUser <# "SimpleX-Directory> The group ID 1 (privacy) is listed (SimpleX-Directory role is changed to admin)."
+        groupFound cath "privacy"
+        -- de-listed if owner role changed
+        cath ##> "/mr privacy bob admin"
+        cath <## "#privacy: you changed the role of bob from owner to admin"
+        bob <## "#privacy: cath changed your role from owner to admin"
+        bob <# "SimpleX-Directory> Your role in the group ID 1 (privacy) is changed to admin."
+        bob <## ""
+        bob <## "The group is no longer listed in the directory."
```

Comment:
service could also send a message to the group so that member who changes role of registrator learns this results in group being de-listed. Same for profile change, member removal, etc.

Reply:
interesting, but probably unnecessary, as only owners can do it, and usually there is only one owner, and if not - they can say it in the group... Also, in the future approval should be automated, so no drama...